icicle/tests/test_hash_api.cpp

```diff
@@ -134,6 +247,7 @@ class HashSumBackend : public HashBackend
 
 TEST_F(HashApiTest, MerkleTree)
 {
+  ICICLE_CHECK(icicle_set_device(s_reference_target)); // TODO CUDA too
```

Comment:
what about the TODO

Reply:
there is no merkle tree for GPU yet so the TODO is for Roman I guess :)